jerry-core/jerry-snapshot.c

```diff
@@ -630,3 +632,409 @@ jerry_exec_snapshot (const void *snapshot_p, /**< snapshot */
+  bytes_copied = append_chars_to_buffer (buffer_p,
+                                         buffer_end_p,
+                                         total_bytes,
+                                         (const char *) str_buffer_p,
+                                         str_buffer_size);
+
+  ECMA_FINALIZE_UTF8_STRING (str_buffer_p, str_buffer_size);
+
+  return bytes_copied;
+} /* append_ecma_string_to_buffer */
+
+/**
+ * Append an unsigned number to the specified buffer.
+ *
+ * @return number of bytes, actually copied to the buffer.
+ */
+static lit_utf8_size_t
+append_number_to_buffer (uint8_t *buffer_p, /**< buffer */
+                         uint8_t *buffer_end_p, /**< the end of the buffer */
+                         size_t *total_bytes, /**< copied bytes */
+                         lit_utf8_size_t number) /**< number */
+{
+  lit_utf8_byte_t uint32_to_str_buffer[ECMA_MAX_CHARS_IN_STRINGIFIED_UINT32];
+  lit_utf8_size_t utf8_str_size = ecma_uint32_to_utf8_string (number,
+                                                              uint32_to_str_buffer,
+                                                              ECMA_MAX_CHARS_IN_STRINGIFIED_UINT32);
+
+  JERRY_ASSERT (utf8_str_size <= ECMA_MAX_CHARS_IN_STRINGIFIED_UINT32);
+
+  lit_utf8_size_t bytes_copied = append_chars_to_buffer (buffer_p,
+                                                         buffer_end_p,
+                                                         total_bytes,
+                                                         (const char *) uint32_to_str_buffer,
+                                                         utf8_str_size);
+  return bytes_copied;
+} /* append_number_to_buffer */
+
+/**
+ * Check whether the passed ecma-string is a valid identifier.
+ *
+ * @return true, if the ecma-string is a valid identifier,
+ *         false - otherwise.
+ */
+static bool
+ecma_string_is_valid_identifier (ecma_string_t *string_p)
+{
+  bool result = false;
+
+  ECMA_STRING_TO_UTF8_STRING (string_p, str_buffer_p, str_buffer_size);
+
+  if (lit_char_is_identifier_start (str_buffer_p))
+  {
+    const uint8_t *str_start_p = str_buffer_p;
+    const uint8_t *str_end_p = str_buffer_p + str_buffer_size;
+
+    result = true;
+
+    while (str_start_p < str_end_p)
+    {
+      if (!lit_char_is_identifier_part (str_start_p))
+      {
+        result = false;
+        break;
+      }
+      lit_utf8_incr (&str_start_p);
+    }
+  }
+  else
+  {
+    result = false;
+  }
+
+  ECMA_FINALIZE_UTF8_STRING (str_buffer_p, str_buffer_size);
+
+  return result;
+} /* ecma_string_is_valid_identifier */
+
+#endif /* JERRY_ENABLE_SNAPSHOT_SAVE */
+
+/**
+ * Copy certain string literals into the given buffer in a specified format,
+ * which are valid identifiers and none of them are magic string.
+ *
+ * @return size of the literal-list in bytes, at most equal to the buffer size,
+ *          if the source parsed successfully and the list of the literals isn't empty,
```

Comment:
At `jerry_parse_and_save_snapshot` there are the same style, which one is nicer?